app/src/main/java/com/example/vshcheglov/webshop/presentation/login/LoginViewModel.kt

```diff
@@ -1,55 +1,69 @@
 package com.example.vshcheglov.webshop.presentation.login
 
 import androidx.biometric.BiometricPrompt
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import com.example.vshcheglov.webshop.App
 import com.example.vshcheglov.webshop.data.DataProvider
 import com.example.vshcheglov.webshop.domain.User.UserCredentials
 import com.example.vshcheglov.webshop.extensions.isEmailValid
 import com.example.vshcheglov.webshop.extensions.isPasswordValid
 import com.example.vshcheglov.webshop.presentation.helpres.Encryptor
+import com.example.vshcheglov.webshop.presentation.helpres.EventWithContent
 import kotlinx.coroutines.*
-import nucleus5.presenter.Presenter
 import timber.log.Timber
 import java.lang.Exception
 import javax.crypto.Cipher
 import javax.inject.Inject
 
-class LoginPresenter : Presenter<LoginPresenter.View>() {
+class LoginViewModel : ViewModel() {
 
     @Inject
     lateinit var dataProvider: DataProvider
     @Inject
     lateinit var encryptor: Encryptor
 
-    private lateinit var job: Job
-    private lateinit var uiCoroutineScope: CoroutineScope
+    private val job: Job = Job()
+    private val uiCoroutineScope: CoroutineScope = CoroutineScope(Dispatchers.Main + job)
+
+    private val _stateLiveData = MutableLiveData<LoginViewState>().apply {
+        value = LoginViewState()
+    }
+    val stateLiveData: LiveData<LoginViewState> = _stateLiveData
+
+    private val _commandLiveData: MutableLiveData<EventWithContent<LoginCommand>> = MutableLiveData()
+    val commandLiveData: LiveData<EventWithContent<LoginCommand>> = _commandLiveData
+
+    private fun getState() = stateLiveData.value!!
 
     init {
         App.appComponent.inject(this)
+        initEmailFromCredentials()
     }
 
     fun logInUser(email: String, password: String, isNetworkAvailable: Boolean) {
         var isValid = true
         if (!email.isEmailValid()) {
```

Comment:
`isEmailValid ` - this should not be string extension. Better to use as StringUtils class